inc/classes/Addons/Cloudflare/Cloudflare.php

```diff
@@ -0,0 +1,611 @@
+<?php
+namespace WP_Rocket\Addons\Cloudflare;
+
+use WP_Rocket\Admin\Options_Data;
+
+/**
+ * Cloudflare
+ *
+ * @since 3.5
+ * @author Soponar Cristina
+ */
+class Cloudflare {
+	/**
+	 * WP Rocket options instance.
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Cloudflare API instance or WP_Error
+	 *
+	 * @var \Cloudflare\Api
+	 */
+	private $cloudflare_api;
+
+	/**
+	 * Cloudflare User Agent
+	 */
+	const CF_USER_AGENT = 'wp-rocket/' . WP_ROCKET_VERSION;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Options_Data $options WP Rocket options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options        = $options;
+		$this->cloudflare_api = $this->get_cloudflare_instance();
```

Comment:
If  you do that at the creation of the instance, you have no check to make sure the CF addon is enabled. That would send API requests and display the error for no reason.

Reply:
It shouldn't display any errors because everything is behind the do_cloudflare check ( https://www.screencast.com/t/BEwiVPEz1 ) . The validation API credentials is made on the constructor and saved in a transient. The transient is deleted / updated in 2 situations and expires in 2 weeks:
1. When Cloudflare addon is disabled / enabled it will delete the transient 
2. When the CF credentials are changed it will revalidate the credentials.
Not sure if I'm missing something in here :)

Reply:
The check in the constructor will be good yes